### PR TITLE
Presentation: Hierarchy level size limiting

### PR DESCRIPTION
**Pull request by grigasp**
Part of https://github.com/iTwin/itwinjs-backlog/issues/104
Related to https://github.com/iTwin/imodel-native/pull/187 and https://github.com/iTwin/presentation/pull/89
imodel-native: https://github.com/iTwin/imodel-native/tree/presentation/hierarchy-level-size-limiting

### CODE REVIEW

File: docs/presentation/hierarchies/FilteringLimiting.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+1. Hierarchy levels that support filtering have a filter icon at the right of the parent node when it's hovered:
+
+   ![Filter icon on filterable parent node](./media/filtering-limiting-workflow-1.png)
+
+2. When the button is pressed, the filtering dialog shows up:
+
+   ![Filtering dialog](./media/filtering-limiting-workflow-2.png)
+
+   The dialog allows filtering the results by class and properties of elements w hich the hierarchy level consists of and doesn't display properties of classes that don't exist in the hierarchy level. In addition, selecting a class filters the available property list to show only properties for that class of elements. And it works in the opposite direction too - selecting a property filters the classes to show only the ones that have the property.
```
Comment (saskliutas) [inline docs/presentation/hierarchies/FilteringLimiting.md:27]:
```suggestion
   The dialog allows filtering the results by class and properties of elements which the hierarchy level consists of and doesn't display properties of classes that don't exist in the hierarchy level. In addition, selecting a class filters the available property list to show only properties for that class of elements. And it works in the opposite direction too - selecting a property filters the classes to show only the ones that have the property.
```

File: docs/presentation/hierarchies/FilteringLimiting.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+```tsx
+function MyTree(props: MyTreeProps) {
+  const { nodeLoader, onItemsRendered } = usePresentationTreeNodeLoader({
+    // supply nodes loader props
+  });
+  const eventHandler = useUnifiedSelectionTreeEventHandler({ nodeLoader });
+  const treeModel = useTreeModel(nodeLoader.modelSource);
+
+  // create presentation-specific tree renderer
+  const treeRenderer = React.useCallback(
```
Comment (saskliutas) [inline docs/presentation/hierarchies/FilteringLimiting.md:66]: I don't think there is a need to have a stable reference of `treeRenderer` so `useCallback` can be removed to simplify this example.

**Review by saskliutas — APPROVED** [review]

File: docs/presentation/hierarchies/FilteringLimiting.md (hunk 2 above)
Comment (grigasp) [reply docs/presentation/hierarchies/FilteringLimiting.md]: simplified

File: docs/changehistory/NextVersion.md [hunk 3]
```diff
@@ -50,4 +50,23 @@ The lower left is the original (smaller, inside) mesh with the (transparent) off
...
\ No newline at end of file
+There is also new support for intersecting a `Ray3d` with a triangle or a polygon. [BarycentricTriangle.intersectRay3d]($core-geometry) and [BarycentricTriangle.intersectSegment]($core-geometry) return a [TriangleLocationDetail]($core-geometry) for the intersection point of the plane of the triangle with the infinite line parameterized by a ray or segment. Similarly, [PolygonOps.intersectRay3d]($core-geometry) returns a [PolygonLocationDetail]($core-geometry) for the intersection point in the plane of the polygon. Both returned detail objects contain properties classifying where the intersection point lies with respect to the triangle/polygon, including `isInsideOrOn` and closest edge data.
+
+## Presentation
```
Comment (pmconne) [inline docs/changehistory/NextVersion.md:55]: Add to table of contents.

**Review by pmconne — APPROVED** [review]

File: docs/changehistory/NextVersion.md (hunk 3 above)
Comment (grigasp) [reply docs/changehistory/NextVersion.md:58]: added

**Comment by grigasp** [comment]
I skipped a failing test that needs to be updated to match the latest addon behavior. @MichaelBelousov is aware.

**Review by MichaelBelousov — APPROVED** [review]

**Pull request merged**